Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.mm

```diff
@@ -109,19 +109,33 @@
     for (auto& currNode : stateTree.nodeMap().values()) {
         if (currNode->hasChangedProperty(ScrollingStateNode::Property::Layer)) {
             auto platformLayerID = PlatformLayerID { currNode->layer() };
-            currNode->setLayer(layerTreeHost.layerForID(platformLayerID));
+            auto remoteLayerTreeNode = layerTreeHost.nodeForID(platformLayerID);
+            if (remoteLayerTreeNode) {
+                currNode->setLayer(remoteLayerTreeNode->layer());
+#if ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION)
+                currNode->setInteractionRegionsLayer(remoteLayerTreeNode->interactionRegionsLayer());
+#endif
+            }
 #if ENABLE(OVERLAY_REGIONS_IN_EVENT_REGION)
             if (platformLayerID && (currNode->isFixedNode() || currNode->isStickyNode()))
                 m_fixedScrollingNodeLayerIDs.add(platformLayerID);
 #endif
         }
-        
+
         switch (currNode->nodeType()) {
         case ScrollingNodeType::Overflow: {
             ScrollingStateOverflowScrollingNode& scrollingStateNode = downcast<ScrollingStateOverflowScrollingNode>(*currNode);
 
-            if (scrollingStateNode.hasChangedProperty(ScrollingStateNode::Property::ScrollContainerLayer))
-                scrollingStateNode.setScrollContainerLayer(layerTreeHost.layerForID(PlatformLayerID { scrollingStateNode.scrollContainerLayer() }));
+            if (scrollingStateNode.hasChangedProperty(ScrollingStateNode::Property::ScrollContainerLayer)) {
+                auto platformLayerID = PlatformLayerID { scrollingStateNode.scrollContainerLayer() };
+                auto remoteLayerTreeNode = layerTreeHost.nodeForID(platformLayerID);
+                if (remoteLayerTreeNode) {
+                    scrollingStateNode.setScrollContainerLayer(remoteLayerTreeNode->layer());
+#if ENABLE(INTERACTION_REGIONS_IN_EVENT_REGION)
+                    scrollingStateNode.setInteractionRegionsLayer(remoteLayerTreeNode->interactionRegionsLayer());
```

Comment:
I don't know that you need to call this here; this is the "scroll container" code path. The call below `currNode->setLayer()` should be sufficient?

Reply:
double checking, but I think I needed it to sync in `ScrollingTreeScrollingNodeDelegateIOS ::repositionScrollingLayers()` for `overflow:scroll` and iframes (below)